src/main/java/org/folio/circulation/domain/Account.java

```diff
@@ -1,50 +1,196 @@
 package org.folio.circulation.domain;
 
-import static org.folio.circulation.support.JsonPropertyFetcher.getNestedStringProperty;
-import static org.folio.circulation.support.JsonPropertyFetcher.getProperty;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.folio.circulation.support.JsonPropertyWriter;
+import org.joda.time.DateTime;
 
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Comparator;
 import java.util.Optional;
+import java.util.UUID;
 
-import org.apache.commons.lang3.math.NumberUtils;
-import org.joda.time.DateTime;
-
-import io.vertx.core.json.JsonObject;
+import static org.folio.circulation.support.JsonPropertyFetcher.getNestedStringProperty;
+import static org.folio.circulation.support.JsonPropertyFetcher.getProperty;
 
 public class Account {
+  private final String id;
+  private final String ownerId;
+  private final String feeFineId;
+  private final Double amount;
+  private final Double remaining;
+  private final String feeFineType;
+  private final String feeFineOwner;
+  private final String title;
+  private final String barcode;
+  private final String callNumber;
+  private final String location;
+  private final String materialTypeId;
+  private final String loanId;
+  private final String userId;
+  private final String itemId;
+  private final String status;
+  private final String paymentStatus;
 
-  private final JsonObject representation;
   private Collection<FeeFineAction> feeFineActions = new ArrayList<>();
 
+  public Account(Loan loan, Item item, FeeFineOwner feeFineOwner,
+    FeeFine feeFine, Double amount) {
+    this.id = UUID.randomUUID().toString();
+    this.ownerId = feeFineOwner.getId();
+    this.feeFineId = feeFine.getId();
+    this.amount = amount;
+    this.remaining = amount;
+    this.feeFineType = feeFine.getFeeFineType();
+    this.feeFineOwner = feeFineOwner.getOwner();
+    this.title = item.getTitle();
+    this.barcode = item.getBarcode();
+    this.callNumber = item.getCallNumber();
+    this.location = item.getLocation().getPrimaryServicePointId().toString();
+    this.materialTypeId = item.getMaterialTypeId();
+    this.loanId = loan.getId();
+    this.userId = loan.getUserId();
+    this.itemId = item.getItemId();
+    this.status = "Open";
+    this.paymentStatus = "Outstanding";
+  }
+
   public Account(JsonObject representation) {
```

Comment:
Personally, I would make all of the methods that use JsonObject for instantiation into static factory methods, and use constructors that don't rely on JsonObject, meaning that the JsonObject cannot be used in the internal state.

Reply:
Done.